File: app/src/main/java/tech/michaeloverman/mscount/dataentry/NoteValueAdapter.java
package tech.michaeloverman.mscount.dataentry;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.TypedArray;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

import tech.michaeloverman.mscount.R;
import tech.michaeloverman.mscount.pojos.PieceOfMusic;
import timber.log.Timber;

/**
 * Adapter class handles the recycler view which provides options for baseline rhythmic values.
 *
 * Created by Michael on 5/24/2017.
 */
class NoteValueAdapter extends RecyclerView.Adapter<NoteValueAdapter.NoteViewHolder> {
    private final Context mContext;
    private final TypedArray noteValueImages;
    private final String[] noteValueDescriptions;
    private int selectedPosition;

    public NoteValueAdapter(Context context, TypedArray images, String[] imageDescriptions) {
        mContext = context;
        noteValueImages = images;
        noteValueDescriptions = imageDescriptions;
    }

    public int getSelectedRhythm() {
        switch(selectedPosition) {
            case 0: return PieceOfMusic.SIXTEENTH;
            case 1: return PieceOfMusic.DOTTED_SIXTEENTH;
            case 2: return PieceOfMusic.EIGHTH;
            case 3: return PieceOfMusic.DOTTED_EIGHTH;
            case 5: return PieceOfMusic.DOTTED_QUARTER;
            case 6: return PieceOfMusic.HALF;
            case 7: return PieceOfMusic.DOTTED_HALF;
            case 8: return PieceOfMusic.WHOLE;
            case 4:
            default: return PieceOfMusic.QUARTER;
        }
    }

    public void setSelectedPosition(int rhythm) {
        Timber.d("setting selected rhythmic value: %s", rhythm);
        switch(rhythm) {
            case PieceOfMusic.SIXTEENTH:
                selectedPosition = 0;
                break;
            case PieceOfMusic.DOTTED_SIXTEENTH:
                selectedPosition = 1;
                break;
            case PieceOfMusic.EIGHTH:
                selectedPosition = 2;
                break;
            case PieceOfMusic.DOTTED_EIGHTH:
                selectedPosition = 3;
                break;
            case PieceOfMusic.DOTTED_QUARTER:
                selectedPosition = 5;
                break;
            case PieceOfMusic.HALF:
                selectedPosition = 6;
                break;
            case PieceOfMusic.DOTTED_HALF:
                selectedPosition = 7;
                break;
            case PieceOfMusic.WHOLE:
                selectedPosition = 8;
                break;
            case PieceOfMusic.QUARTER:
            default: selectedPosition = 4;
        }
        Timber.d("selectedPosition = %s", selectedPosition);
    }

    @NotNull
    @Override
    public NoteViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        View item = LayoutInflater.from(mContext)
                .inflate(R.layout.note_value_image_view, parent, false);
        return new NoteViewHolder(item);
    }

    @SuppressLint("RecyclerView")
    @Override
    public void onBindViewHolder(NoteViewHolder holder, final int position) {
        holder.image.setImageDrawable(noteValueImages.getDrawable(position));
        Timber.d("onBindViewHolder, position: " + position + " selected: " + (position == selectedPosition));

        if(selectedPosition == position) {
            holder.itemView.setBackground(ContextCompat
                    .getDrawable(mContext, R.drawable.roundcorner_accent));
        } else {
            holder.itemView.setBackground(ContextCompat
                    .getDrawable(mContext, R.drawable.roundcorner_parchment));
        }

        holder.itemView.setOnClickListener( (View v) -> {
            notifyItemChanged(selectedPosition);
            selectedPosition = position;
            notifyItemChanged(selectedPosition);
        });

        holder.itemView.setContentDescription(noteValueDescriptions[position]);
    }

    @Override
    public int getItemCount() {
        return noteValueImages.length();
    }

    class NoteViewHolder extends RecyclerView.ViewHolder {
        final ImageView image;

        public NoteViewHolder(View itemView) {
            super(itemView);
            image = itemView.findViewById(R.id.note_value_image);
            Timber.d("NoteViewHolder created, image: ");
        }
    }
}

File: app/src/main/java/tech/michaeloverman/mscount/dataentry/NoteValueAdapter.kt
package tech.michaeloverman.mscount.dataentry

import android.annotation.SuppressLint
import android.content.Context
import android.content.res.TypedArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.note_value_image_view.view.*
import tech.michaeloverman.mscount.R
import tech.michaeloverman.mscount.dataentry.NoteValueAdapter.NoteViewHolder
import tech.michaeloverman.mscount.pojos.PieceOfMusic
import timber.log.Timber

/**
 * Adapter class handles the recycler view which provides options for baseline rhythmic values.
 *
 * Created by Michael on 5/24/2017.
 */
internal class NoteValueAdapter(private val mContext: Context, private val noteValueImages: TypedArray, private val noteValueDescriptions: Array<String>) : RecyclerView.Adapter<NoteViewHolder>() {
    private var selectedPosition = 0
    val selectedRhythm: Int
        get() = when (selectedPosition) {
            0 -> PieceOfMusic.SIXTEENTH
            1 -> PieceOfMusic.DOTTED_SIXTEENTH
            2 -> PieceOfMusic.EIGHTH
            3 -> PieceOfMusic.DOTTED_EIGHTH
            5 -> PieceOfMusic.DOTTED_QUARTER
            6 -> PieceOfMusic.HALF
            7 -> PieceOfMusic.DOTTED_HALF
            8 -> PieceOfMusic.WHOLE
            4 -> PieceOfMusic.QUARTER
            else -> PieceOfMusic.QUARTER
        }

    fun setSelectedPosition(rhythm: Int) {
        Timber.d("setting selected rhythmic value: %s", rhythm)
        selectedPosition = when (rhythm) {
            PieceOfMusic.SIXTEENTH -> 0
            PieceOfMusic.DOTTED_SIXTEENTH -> 1
            PieceOfMusic.EIGHTH -> 2
            PieceOfMusic.DOTTED_EIGHTH -> 3
            PieceOfMusic.DOTTED_QUARTER -> 5
            PieceOfMusic.HALF -> 6
            PieceOfMusic.DOTTED_HALF -> 7
            PieceOfMusic.WHOLE -> 8
            PieceOfMusic.QUARTER -> 4
            else -> 4
        }
        Timber.d("selectedPosition = %s", selectedPosition)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
        val item = LayoutInflater.from(mContext)
                .inflate(R.layout.note_value_image_view, parent, false)
        return NoteViewHolder(item)
    }

    @SuppressLint("RecyclerView")
    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
        holder.image.setImageDrawable(noteValueImages.getDrawable(position))
        Timber.d("onBindViewHolder, position: " + position + " selected: " + (position == selectedPosition))
        if (selectedPosition == position) {
            holder.itemView.background = ContextCompat
                    .getDrawable(mContext, R.drawable.roundcorner_accent)
        } else {
            holder.itemView.background = ContextCompat
                    .getDrawable(mContext, R.drawable.roundcorner_parchment)
        }
        holder.itemView.setOnClickListener { v: View? ->
            notifyItemChanged(selectedPosition)
            selectedPosition = position
            notifyItemChanged(selectedPosition)
        }
        holder.itemView.contentDescription = noteValueDescriptions[position]
    }

    override fun getItemCount(): Int {
        return noteValueImages.length()
    }

    internal inner class NoteViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val image: ImageView = itemView.note_value_image

        init {
            Timber.d("NoteViewHolder created, image: ")
        }
    }
}